cycle.go

```diff
@@ -91,15 +92,16 @@ func detectCycles(n provider, c containerStore, path []cycleEntry) error {
 
 		entry := cycleEntry{Func: n.Location(), Key: k}
 
-		for _, p := range path {
-			if p.Key == k {
-				err = errCycleDetected{Path: append(path, entry)}
-				return false
-			}
+		// The first element of path is the new addition to the graph, therefore
+		// it must be in any cycle that exists, assuming verifyAcyclic has been
+		// run for every previous Provide.
+		if path[0].Key == k {
```

Comment:
nice catch.

I would add a comment reminding readers that path will always have at least one item.

Reply:
I will add that @abhinav . @zmt do you mean the first addition to the graph overall? `path[0]` here will only ever be a **result** from the function `Provide`d. Not sure if that answers your question.